### PR TITLE
Feature/54 implement doctor update possibility

### PR DESCRIPTION
**Pull request by edward0802**

### CODE REVIEW

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 1]
```diff
@@ -104,5 +113,52 @@ private IQueryable<DoctorModel> PaginationHelper(IQueryable<Doctor> doctors, int
...
+        {
+            var addedTime = DateTime.Now;
+            var result = true;
+            var doctor = _context.Users
+                .Include(u => u.Doctor)
+                .FirstOrDefault(u => u.Id == userId);
+            if (doctor == null)
+            {
+                result = false;
+                return result;
```
Comment (eugent482) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: I think "result" variable is unnecessary. You can return "false" here and "true" at the end.

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 2]
```diff
@@ -50,5 +54,60 @@ public IEnumerable<SpecializationModel> GetSpecializations()
...
+            }
+            return BadRequest();
+        }
+
+        [HttpGet("getProfile")]
+        [Authorize]
+        public IActionResult GetDoctorProfile()
+        {
+
+            var patientId = User.Identity?.Name;
```
Comment (eugent482) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: unused/unnecessary variable

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 3]
```diff
@@ -50,5 +54,60 @@ public IEnumerable<SpecializationModel> GetSpecializations()
         {
             return _doctorService.GetAvailableSpecialization();
         }
+
+        [HttpPut("update")]
+        [Authorize]
```
Comment (eugent482) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: I think would be better to make role-based authorize (only for doctors).

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 4]
```diff
@@ -50,5 +54,60 @@ public IEnumerable<SpecializationModel> GetSpecializations()
...
+                else
+                {
+                    return BadRequest(new { message = "Error with user ID!" });
+                }
+            }
+            return BadRequest();
+        }
+
+        [HttpGet("getProfile")]
+        [Authorize]
```
Comment (eugent482) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: Role-based authorize is necessary here. Here will be NullReferenceException if patient call this method.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 5]
```diff
@@ -104,5 +113,52 @@ private IQueryable<DoctorModel> PaginationHelper(IQueryable<Doctor> doctors, int
...
+            }
+
+            var temporaryPatient = Mapper.Map<TemporaryUser>(doctorModel);
+            temporaryPatient.AddedTime = addedTime;
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: It can be in single Mapper.Map() method. Watch feed back service for example.

**Review by eugent482 — APPROVED** [review]

**Review by SergiyKozachok — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 6]
```diff
@@ -6,16 +6,25 @@
...
         private readonly ApplicationContext _context;
+        private readonly UserManager<User> _userManager;
+        private readonly IFilesService _uploadingFiles;
+        const string DOCTOR = "Doctor";
 
-        public DoctorService(ApplicationContext context)
+        public DoctorService(ApplicationContext context, UserManager<User> userManager, IFilesService uploadingFiles)
         {
             _context = context;
+            _userManager = userManager;
```
Comment (martyniukroman) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: _userManeger isn't used

File: InternetHospital.BusinessLogic/Services/FilesService.cs [hunk 7]
```diff
@@ -16,17 +16,22 @@ public class FilesService : IFilesService
...
         const int MAX_WIDTH = 3000;
         const int IMAGE_MAX_LENGTH = 20;
+        const string PASSPORT = "Passport";
+        const string DIPLOMA = "Diploma";
+        const string LICENSE = "License";
 
-        public FilesService(IHostingEnvironment env, UserManager<User> userManager)
+        public FilesService(IHostingEnvironment env, UserManager<User> userManager, ApplicationContext context)
         {
             _env = env;
```
Comment (martyniukroman) [inline InternetHospital.BusinessLogic/Services/FilesService.cs:32]: _env isn't used

File: InternetHospital.BusinessLogic/Services/FilesService.cs [hunk 8]
```diff
@@ -16,17 +16,22 @@ public class FilesService : IFilesService
...
             _userManager = userManager;
```
Comment (martyniukroman) [inline InternetHospital.BusinessLogic/Services/FilesService.cs:33]: _userManager isn't used

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 9]
```diff
@@ -50,5 +54,58 @@ public IEnumerable<SpecializationModel> GetSpecializations()
...
+            if (!int.TryParse(User.Identity.Name, out int userId))
+            {
+                return BadRequest();
+            }
+
+            var doctor =  _doctorService.GetDoctorProfile(userId);
+            if (doctor != null)
+            {
+                return Ok(doctor);
+
```
Comment (martyniukroman) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: this empty line isn't needed

File: InternetHospital.WebApi/Controllers/PatientController.cs [hunk 10]
```diff
@@ -84,15 +76,34 @@ public async Task<IActionResult> UpdatePatientProfile([FromForm(Name = "Passport
...
+            if (!int.TryParse(User.Identity.Name, out int userId))
+            {
+                return BadRequest();
+            }
+
+            var patient = await _patientService.GetPatientProfile(userId);
+            if (patient != null)
+            {
+                return Ok(patient);
+
```
Comment (martyniukroman) [inline InternetHospital.WebApi/Controllers/PatientController.cs]: copy-paste detected (remove empty line)

File: InternetHospital.BusinessLogic/Services/FilesService.cs [hunk 11]
```diff
@@ -66,7 +71,34 @@ public async Task<User> UploadAvatar(IFormFile image, User user)
...
+            if (images.Count == 0)
+            {
+                return user;
+            }
+            return await UploadFiles(images, user, addedTime, PASSPORT);
+        }
+
+        public async Task<User> UploadDiploma(IFormFileCollection images, User user, DateTime addedTime)
+        {
+            if (images.Count == 0)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/FilesService.cs]: Can it be simplified by ternary operator?

File: InternetHospital.BusinessLogic/Services/FilesService.cs [hunk 12]
```diff
@@ -66,7 +71,34 @@ public async Task<User> UploadAvatar(IFormFile image, User user)
             return user;
         }
 
-        public async Task<User> UploadPassport(IFormFileCollection images, User user)
+        public async Task<User> UploadPassport(IFormFileCollection images, User user, DateTime addedTime)
+        {
+            if (images.Count == 0)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/FilesService.cs]: Can it be simplified by ternary operator?

File: InternetHospital.BusinessLogic/Services/FilesService.cs [hunk 13]
```diff
@@ -66,7 +71,34 @@ public async Task<User> UploadAvatar(IFormFile image, User user)
...
+            {
+                return user;
+            }
+            return await UploadFiles(images, user, addedTime, DIPLOMA);
+        }
+
+        public async Task<User> UploadLicense(IFormFileCollection images, User user, DateTime addedTime)
+        {
+            if (images.Count == 0)
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/FilesService.cs]: Here too.

File: InternetHospital.BusinessLogic/Services/FilesService.cs (hunk 7 above)
Comment (edward0802) [reply InternetHospital.BusinessLogic/Services/FilesService.cs:32]: be attentive, please. this variable is being used

File: InternetHospital.BusinessLogic/Services/FilesService.cs (hunk 8 above)
Comment (edward0802) [reply InternetHospital.BusinessLogic/Services/FilesService.cs:33]: be attentive, please. this variable is being used!

Comment (martyniukroman) [reply InternetHospital.BusinessLogic/Services/FilesService.cs:33]: sorry my fault

File: InternetHospital.BusinessLogic/Services/FilesService.cs (hunk 7 above)
Comment (martyniukroman) [reply InternetHospital.BusinessLogic/Services/FilesService.cs:32]: sorry my fault

**Review by martyniukroman — APPROVED** [review]

**Review by serhiidotsyk — APPROVED** [review]

**Pull request merged**